examples/misc/multiprocess.py

```diff
@@ -42,14 +41,16 @@ def call_back():
             self.fig.canvas.draw()
             return True
 
-        return call_back
+        return call_back()
 
     def __call__(self, pipe):
         print('starting plotter...')
 
         self.pipe = pipe
         self.fig, self.ax = plt.subplots()
-        self.gid = gobject.timeout_add(1000, self.poll_draw())
+        timer = self.fig.canvas.new_timer(interval=1000)
+        timer.add_callback(self.poll_draw)
```

Comment:
I would just lift `call_back` to being a method of `ProcessPlotter` and do `timer.add_callback(self.call_back)`.